manifests/params.pp

```diff
@@ -179,13 +192,16 @@
 
   # Passenger config
   $server_app_root = "${dir}/rack"
-  $server_ssl_dir  = "${vardir}/ssl"
+  $server_ssl_dir  = $ssldir
 
   $server_package     = undef
   $server_version     = undef
-  $client_package     = $::osfamily ? {
-    'Debian' => ['puppet-common','puppet'],
-    default  => ['puppet'],
+  $client_package     = versioncmp($::puppetversion, '4.0') ? {
+    '-1'     => $::osfamily ? {
+      'debian' => ['puppet-common','puppet'],
+      default   => ['puppet'],
+    },
+    default => 'puppet-agent',
```

Comment:
Fedora 22 already has a puppet 4 package non-AIO style and for FreeBSD I could make one quite quickly, but it would be limited to 4.1, as 4.2 requires the C++ rewrite of facter.

I'd just be happier to have parameters set by puppetversion, where there's a real difference because of the version, and another variable that's used to set things according to the packaging style used (AIO/PE (when needed)/native).